compiler/src/dmd/chkformat.d

```diff
@@ -860,34 +886,30 @@ Format parseGenericFormatSpecifier(scope const char[] format,
     {
         case 'd':
         case 'i':
-            if (lm == 'L')
-                specifier = Format.error;
-            else
-                specifier = lm == 'h' && lm2 ? Format.hhd :
-                            lm == 'h'        ? Format.hd  :
-                            lm == 'l' && lm2 ? Format.lld :
-                            lm == 'l'        ? Format.ld  :
-                            lm == 'j'        ? Format.jd  :
-                            lm == 'z'        ? Format.zd  :
-                            lm == 't'        ? Format.td  :
-                                               Format.d;
+            specifier = flags == Modifier.none ? Format.d   :
```

Comment:
Do I miss something If I suggest to switch on flags instead of cascading all these CondExp ?

Reply:
Yes, you are assuming the compiler is [not smarter than you](https://godbolt.org/z/vja3Wx41K). :-)

These conditionals are also briefer than a switch.

Reply:
nice optim from GDC side but DMD does not do so, also even with GDV this would allow to reduce the pressure on the optimizer. But well it's just a detail.

Reply:
I'm not under the impression that anyone thinks dmd is an optimizing compiler.

Reply:
The optimizer is a (very big) bag of heuristics, with exception to dce (and dse to some negligible effect), there's not really much in the way of easing pressure.